### ISSUE TITLE
New fields requested for future analysis

### ISSUE THREAD

**Issue by marshalc** [ouh-churchill/COPE#336]
From @ij-cope - Feb 7th, 2019
```
During the past months Lucy and I have identified the following that – when implemented in the database – would greatly improve any future work done.
 
We have obtained 2 new variables that would need to be added
 
Donor hypertension: yes / no / unknown (to be added to the donor form)
PRA of the recipient at time of transplant: this is a continuous variable in percentage (to be added to the recipient form)
 
Also, it would be essential to have variables in the database that reflect whether a case is eligible for the intention to treat analysis
Would it be possible to add following variables (perhaps in a separate form)
 
Eligible for ITT analysis: Y / N
Actual treatment received: HMPO2, HMP, CS, unkown
```

**Comment by marshalc** [comment]
Donor hypertension - can you give me a sense of where this fits into the range of existing questions for the form please?

Is this a [ Yes, No, Unknown, Not Answered ] question or just a [ Yes, No, Not Answered ] one?

And what will be the default answer for when this is introduced? (as all existing records will be set to this value initially)

**Comment by marshalc** [comment]
"PRA"? - Sorry, that TLA is not registering on me at the moment. Can you expand please?

This will be an integer numeric field? (i.e. no decimals points or fractions of a value?)

What should the default answer be? Null (typical default meaning not answered), or a value from 0-100?

Where would this make sense to fit into the range of questions on the Transplant form?

**Comment by marshalc** [comment]
Is the Intention To Treat (ITT) query one for post procedure record keeping, or is this something that would (in an ideal world) have been answered during the initial data collection?

I am presuming this applies to each Kidney? Or is this on a Recipient by recipient basis? (trying to determine what entity this record would be recorded against)

**Comment by marshalc** [comment]
Actual Treatment Received, would make sense as a final qualified answer from data cleanup and analysis - so not something we'd expect a TT to have done during the initial data collection. So this would be recorded against each Kidney, and by an Admin or Stats user, yes?

In which case, is the complete answer set [ HMP02, HMP, Cold Storage, Unknown, Not Answered ], or should the initial default answer be Unknown, and thus the Not Answered option is redundant?

**Comment by ij-cope** [comment]
Hi Carl

> Donor hypertension - can you give me a sense of where this fits into the range of existing questions for the form please?

_in the "Donor Preop Data" where we currently have "donor diabetes" "alcohol abuse"_

 
> Is this a [ Yes, No, Unknown, Not Answered ] question or just a [ Yes, No, Not Answered ] one?

_Yes, No, Unknown, Not Answered_

> And what will be the default answer for when this is introduced? (as all existing records will be set to this value initially)

_Default to "Not Answered" please_

**Comment by ij-cope** [comment]
Hi Carl

> "PRA"? - Sorry, that TLA is not registering on me at the moment. Can you expand please?

_PRA stands for Panel Reactive Antibodies, which is a test that screens for the amount of antibodies a person has developed against tissue specific antigens_


> This will be an integer numeric field? (i.e. no decimals points or fractions of a value?)

_yes, integer number_

> What should the default answer be? Null (typical default meaning not answered), or a value from 0-100?

_we will have missing values, so if an option "not available" could be there, that would be great_
_values will range between 0 and 100 (in percentage)_
 
> Where would this make sense to fit into the range of questions on the Transplant form?

_it would fit best in "Patient Description" below "Blood Group"_

**Comment by ij-cope** [comment]
Hi Carl

> Is the Intention To Treat (ITT) query one for post procedure record keeping, or is this something that would (in an ideal world) have been answered during the initial data collection?

_for post procedure record keeping. This field details whether or not a kidney has indeed received the treatment it was allocated to and this is something we can only get a handle on in retrospect._
 
> I am presuming this applies to each Kidney? Or is this on a Recipient by recipient basis? (trying to determine what entity this record would be recorded against)

_applies to each kidney individually, yes_

**Comment by ij-cope** [comment]
Hi Carl

> Actual Treatment Received, would make sense as a final qualified answer from data cleanup and analysis - so not something we'd expect a TT to have done during the initial data collection. So this would be recorded against each Kidney, and by an Admin or Stats user, yes?

_Indeed. 
Yes, recorded for each kidney and by an Admin or Stats user_

 
> In which case, is the complete answer set [ HMP02, HMP, Cold Storage, Unknown, Not Answered ], or should the initial default answer be Unknown, and thus the Not Answered option is redundant?

_the not answered question is indeed redundant, thanks to set default to unkonw_

**Comment by marshalc** [comment]
@mertenssarah - There are currently 59 or so registered staff who do not have a location set against them. You can find them by looking at https://cope.nds.ox.ac.uk/en-gb/wp4/staff/?order_by=based_at&ordering=asc and seeing who is missing a "Base". Until these items are corrected I can't apply and database migrations to add new fields to the DB. If unsure of exact location for each person, suggest you use one of the "Other/Country" options based on why that staff member was added.

**Comment by ij-cope** [comment]
Hi @marshalc 

Sarah and I went over this on Friday, all registered staff now have a location.

**Comment by marshalc** [comment]
Thank you @ij-cope & @mertenssarah. 

> _PRA stands for Panel Reactive Antibodies, which is a test that screens for the amount of antibodies a person has developed against tissue specific antigens_
> _it would fit best in "Patient Description" below "Blood Group"_

Adding this to that section would denote this is relevant to both the Donor and the Recipient (as it would be logged against the base Patient record). Is this to appear on both forms for all patients, or is this specific Donor Pre-op data?

**Comment by ij-cope** [comment]
Hi @marshalc 

this is specific pre-op data, thanks

**Comment by marshalc** [comment]
@ij-cope & @ldaviesnds  - how many of these new fields need to feature in the statistician's data extract?

Also, @ij-cope - The ITT and ATR values: Are they to be recorded for every organ in the trial (including those that don't get allocated?) or only for organs that make it through to allocation? This has an impact on whether I can add this to the admin section of the Transplantation Form, or if they will need to be set on their own pages/forms.

**Comment by ldaviesnds** [comment]
Hi Carl,

All new fields that are being added to the database need to feature in the statisticians data extract.

Thanks
Lucy

**Comment by marshalc** [comment]
I had a feeling you'd say that @ldaviesnds . In which case, I have just noticed that where I've added PRA to Lab Results, there is both Last Creatinine and Max Creatinine - but the data extract for you only has Last Creatinine. Do you want me to add the Max Creatinine data as well?

**Comment by ldaviesnds** [comment]
Yes please.

Thanks
Lucy

**Comment by marshalc** [comment]
I think I've partly answered my own question @ij-cope - the ITT and ATR will need their own form, which then begs the question of what data is going to want displaying alongside these two fields in order for them to be answered? I imagine just having the Trial ID and these two fields wouldn't be ideal though is the most straight forward option to build.

**Comment by ij-cope** [comment]
Hi @marshalc 

It is the most straight forward and I am happy with it. 
If those are available in the statisticians report, all we need is there.

The answer to ITT / "actual treatment received" needs us to assess multiple fields that are currently in different forms so it is never straightforward anyway.

thanks

**Comment by marshalc** [comment]
If you have a list of the fields that can be used to assess it, then it may be fairly easy to put them all on the same grid, or at least provide links to the relevant forms.

**Comment by ij-cope** [comment]
These are the essential ones, for each kidney: 

1. for "Eligable for ITT analysis"

organ.transplantable
organ.not_transplantable_reason

organ.donor.multiple_recipients
organ.donor.get_not_randomised_because_display
organ.donor.not_randomised_because_other
organ.donor.randomisation.result
organ.recipient.signed_consent
organ.recipient.single_kidney_transplant

organ.recipient.organ_untransplantable
organ.recipient.organ_untransplantable_reason

2. for "Actual Treatment Received"
This only makes sense to complete in kidneys that are available for the ITT analysis and should be either not available for those with 'N' to 'ITT' or an option 'not applicable' added to ATR (though that would be double data collection in a way).

organ.perfusion_possible
organ.perfusion_not_possible_because
organ.perfusion_started
organ.oxygen_bottle_full
organ.oxygen_bottle_open
organ.oxygen_bottle_changed
organ.oxygen_bottle_changed_at
organ.oxygen_bottle_changed_at_unknown
organ.perfusate_measurable
organ.perfusate_measure
organ.recipient.perfusate_measure
organ.recipient.perfusion_stopped
organ.recipient.organ_cold_stored
organ.recipient.removed_from_machine_at
organ.recipient.oxygen_full_and_open

**Comment by marshalc** [comment]
Many thanks @ij-cope  - I'll see about incorporating those results into the data entry form display.

**Comment by marshalc** [comment]
All enhancements added to codebase, about to attempt deployment on live server.

**Comment by marshalc** [comment]
@ij-cope - This functionality is now live. See Administration Home -> Completeness Data Management -> Post Trial Categorisation  (https://cope.nds.ox.ac.uk/en-gb/wp4/administration/completeness/post-trial/). This provides a list of all Organs in the trial, and each can then be edited for this specific information. The summary data asked for is available in two tabs above the data fields on each page.

**Comment by marshalc** [comment]
PRA field was added to Procurement form (and linked to Donor), when it should have been on the Transplant form (and linked to Recipient). This has been highlighted by Simon K, and agreed at today's conference call that this should be moved (pending a review of any data that has been entered thus far).

**Comment by ij-cope** [comment]
@marshalc - no data has been entered thus far for PRA

**Comment by marshalc** [comment]
From Ina, by email 14-6-19:

The module Post Trial Categorisation is working well.
 
Would you, as a final adjustment, change the question from “Was death treatment related?” to “Which treatment did the kidney receive?”.
 
![image001](https://user-images.githubusercontent.com/572826/59696669-5cfa7680-91e4-11e9-9182-0b4badf6c0a8.jpg)

These data should also be exported in the file for the statistician, should this not already be the case.
 
Thanks
 
Ina

**Comment by marshalc** [comment]
Confirmed, no data was entered in the PRA on the Donor, so when the 0.9.5 release comes through this week, there will be no data to move.

**Comment by marshalc** [comment]
The label issue from Ina is one of region language not updated to reflect changes - this being one of those development tasks that for NL and BE was devolved down to Sarah to review and update.

For example: https://cope.nds.ox.ac.uk/en-gb/wp4/administration/completeness/post-trial/1/ shows
<img width="496" alt="Screenshot 2019-06-18 at 16 24 43" src="https://user-images.githubusercontent.com/572826/59697483-ccbd3100-91e5-11e9-97f4-3b0a8ea937f5.png">

But https://cope.nds.ox.ac.uk/fr-be/wp4/administration/completeness/post-trial/1/  shows
<img width="491" alt="Screenshot 2019-06-18 at 16 26 54" src="https://user-images.githubusercontent.com/572826/59697537-e52d4b80-91e5-11e9-9bd2-3e6f23241047.png">

I'll amend the NL and BE files for this field. As usual, Sarah/Ina will need to check all the other labels use the terminology specific to that region

**Comment by marshalc** [comment]
For @rraajjeevv : Sarah (or one of the EU co-ordinators) would normally edit the relevant `django.po` via the repository. Once they've done that, you'd need to pull the latest changes, and run the `python manage.py compilemessages` and checking that the NL-NL and FR-BE language versions of the site appear as expected. Then you can do an update of the code on the live server and follow the deployment process for that.

One thing to watch out for is that they've edited the relevant latest label files. At the time of writing, this would be the ones on the Dev-0.9.0 branch (i.e. the `django.po` files under https://github.com/ouh-churchill/COPE/tree/Dev-0.9.0/locale/ ), however, this may be more likely to be done on the 'master` branch (as that is the default when viewing the repo).

**Comment by marshalc** [comment]
This issue should now auto-close once the `Dev-0.9.0` branch is merged into `master` for deployment. No further action is anticipated on this issue.
